test/system/executable/CMakeLists.txt

```
@@ -0,0 +1,105 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###########################################################################
+# Create tests                                                            #
+###########################################################################
+
+find_package(PythonInterp 3 REQUIRED)
+
+if(PYTHONINTERP_FOUND)
+
+    # Name of files to test
+    set(TEST_CONFIGURATIONS
+        void_configuration
+        simple_configuration
+    )
+
+    # Copy configuration files
+    # set(CONFIGURATION_FILES
+    #     "configurations/void_configuration.yaml"
+    #     "configurations/simple_configuration.yaml"
+    # )
+    foreach(TEST_CONFIGURATION ${TEST_CONFIGURATIONS})
+        configure_file(${CMAKE_CURRENT_SOURCE_DIR}/configurations/${TEST_CONFIGURATION}.yaml
+            ${CMAKE_CURRENT_BINARY_DIR}/configurations/${TEST_CONFIGURATION}.yaml
+            COPYONLY)
+    endforeach()
+
+    # windows auxiliary script to fork test execution
+    # set(PWS_LAUNCHER
+    #     ${CMAKE_CURRENT_SOURCE_DIR}/launcher.ps1
+    # )
+
+    # Calculate environment
+    set(TEST_ENVIRONMENT
+        "PATH=$ENV{PATH};$<TARGET_FILE_DIR:fastcdr>;$<TARGET_FILE_DIR:fastrtps>;$<TARGET_FILE_DIR:ddsrouter>")
+
+    # if(WIN32)
+
+    #     if(TARGET tinyxml2 OR TARGET tinyxml2::tinyxml2)
+    #         set(TEST_ENVIRONMENT
+    #             "${TEST_ENVIRONMENT};$<TARGET_FILE_DIR:${TINYXML2_LIBRARY}>")
+    #     elseif(EXISTS TINYXML2_LIBRARY)
+    #         get_filename_component(
+    #             TINYXML2_LIBRARY_DIR ${TINYXML2_LIBRARY} DIRECTORY)
+    #         set(TEST_ENVIRONMENT
+    #             "${TEST_ENVIRONMENT};${TINYXML2_LIBRARY_DIR}")
+    #         unset(TINYXML2_LIBRARY_DIR)
+    #     endif()
+
+    #     string(REPLACE ";" "\\;" TEST_ENVIRONMENT "${TEST_ENVIRONMENT}")
+
+    # endif(WIN32)
+
+    # populate the tests
+    foreach(TEST IN LISTS TEST_CONFIGURATIONS)
+
+        set(TEST_NAME "system.ddsrouter.${TEST}")
+
+        # CTest has issues with signals on windows. We forked the test
+        # execution using an auxiliary powershell script
+        if(WIN32)
+        #    add_test(
+        #             NAME system.tools.fastdds.${TEST}
+        #             COMMAND powershell "-File" ${PWS_LAUNCHER}
+        #                     ${PYTHON_EXECUTABLE}
+        #                     ${CMAKE_CURRENT_SOURCE_DIR}/tests.py
+        #                     $<TARGET_FILE:fast-discovery-server>
+        #                     ${TEST}
+        #         )
+        else()
+            add_test(
+                    NAME system.tools.ddsrouter.${TEST}
+                    COMMAND ${PYTHON_EXECUTABLE}
+                            ${CMAKE_CURRENT_SOURCE_DIR}/tests.py
+                            "-e" $<TARGET_FILE:ddsrouter>
+                            "-c" configurations/${TEST}.yaml
+                            # Add -d for debugging
```

I'd use the long version of the args to be more clear and also, as now debug is False by default, set it to True
```suggestion
                            "--exe" $<TARGET_FILE:ddsrouter>
                            "--config-file" configurations/${TEST}.yaml
                            "--debug" # Add -d for debugging
```

---

:+1: